README.md

```diff
@@ -85,932 +119,667 @@ Unlike many authorization systems where a service controls access to resources i
+| `iss`   | `DID`                                     | Yes      | Issuer DID (sender)                                         |
+| `aud`   | `DID`                                     | Yes      | Audience DID (receiver)                                     |
+| `sub`   | `DID`                                     | Yes      | Principal that the chain is about (the [Subject])           |
+| `cmd`   | `String`                                  | Yes      | The [Command] to eventually invoke                          |
+| `args`  | `{String : Any}`                          | Yes      | Any [Arguments] that MUST be present in the Invocation      |
+| `nonce` | `Bytes`                                   | Yes      | Nonce                                                       |
+| `meta`  | `{String : Any}`                          | No       | [Meta] (asserted, signed data) — is not delegated authority |
+| `nbf`   | `Integer` (53-bits[^js-num-size])         | No       | "Not before" UTC Unix Timestamp in seconds (valid from)     |
+| `exp`   | `Integer \| Null` (53-bits[^js-num-size]) | Yes      | Expiration UTC Unix Timestamp in seconds (valid until)      |
 
-Revocations MUST be irreversible. If the revocation was issued in error, a unique UCAN MAY be issued (e.g. by updating the nonce or changing the time bounds). This prevents confusion as the revocation moves through the network and makes revocation stores append-only and highly amenable to caching.
+# Implementation Recommendations
 
-A revocation message MUST conform to the following JSON format:
+## Delegation Store
 
-``` js
-{
-  "iss": did,
-  "revoke": canonicalUcanCid,
-  "challenge": base64Unpadded(sign(did.privateKey, `REVOKE:${canonicalUcanCid}`))
-}
-```
+A validator MAY keep a local store of UCANs that it has received. UCANs are immutable but also time-bound so that this store MAY evict expired or revoked UCANs.
 
-This format makes it easy to select the relevant UCAN, confirm that the issuer is in the proof chain for some or all capabilities, and validate that the revocation was signed by the same issuer.
+This store SHOULD be indexed by CID (content addressing). Multiple indices built on top of this store MAY be used to improve capability search or selection performance.
 
-Any other proofs in the selected UCAN not issued by the same DID as the revocation issuer MUST be treated as valid.
+## Memoized Validation
 
-Revocations MAY be deleted once the UCAN that they reference expires or otherwise becomes invalid via its proactive mechanisms.
+Aside from revocation, capability validation is idempotent. Marking a CID (or capability index inside that CID) as valid acts as memoization, obviating the need to check the entire structure on every validation. This extends to distinct UCANs that share a proof: if the proof was previously reviewed and is not revoked, it is RECOMMENDED to consider it valid immediately.
```

Comment:
Should that by "Aside from revocation **and expiration**,..." ? I am also confused by the corner case mentioned here by Gozala: 

Let's say Alice delegates to Bob a capability `/crud/` over a hard disk, then Bob sends Carol (me) a capability `/crud/write/`.  Before I go to invoke it at Alice's hard disk, Alice's original parent cap expired or was revoked.  Bob sends me a new sub-delegation identical except for having a fresher parent.  That's a different token, right? This idempotency stuff doesn't apply because if it's a different chain, it's a different token?